src/dmd/optimize.d

```diff
@@ -42,24 +42,44 @@ extern (C++) Expression expandVar(int result, VarDeclaration v)
 {
     //printf("expandVar(result = %d, v = %p, %s)\n", result, v, v ? v.toChars() : "null");
 
+    /********
+     * Params:
+     *  e = initializer expression
+     */
+    Expression initializerReturn(Expression e)
+    {
+        if (e.type != v.type)
+        {
+            e = e.castTo(null, v.type);
+        }
+        v.inuse++;
+        e = e.optimize(result);
+        v.inuse--;
+        //if (e) printf("\te = %p, %s, e.type = %d, %s\n", e, e.toChars(), e.type.ty, e.type.toChars());
+        return e;
+    }
+
+    static Expression nullReturn()
+    {
+        return null;
+    }
+
     static Expression errorReturn()
     {
         return new ErrorExp();
     }
 
-    Expression e = null;
     if (!v)
-        return e;
+        return nullReturn();
```

Comment:
It looks like you're just replacing `return null` with function that always returns `null` called "nullReturn".  I'm a little dumbfounded.  What am I missing?

Reply:
It gives me an easy place to hook in things like a printf or other debug code. The inliner will inline it anyway, so it's cost free. And it lines up neatly with the Ddoc description of the 3 things that can be returned.

(When I debug code, I'll often add/remove printf's on the return statements. It's a bit of a hassle when there are multiple return points, and annoying when I miss one.)